### PR TITLE
Remove an assert when WAL sender tries to update a cleared replication slot.

### PR DESCRIPTION
**Pull request by dkovalev1**
Remove an assert when WAL sender tries to update a cleared replication slot.

Due to the asynchronous nature of FTS and WAL communications, there is a
possible situation where the mirror goes down, FTS sends a command to turn off
replication, but the mirror and WAL sender restart before the command from FTS
reaches the segment master. In this case, FTS removes the replication slot on
the master, but the WAL sender tries to update the slot. In this situation,
replication is useless, and WAL sender will exit. 
Assert for this condition looks to be unwanted.

The test max_slot_wal_keep_size.sql hit this rare condition with the Ubuntu
container. This commit removes the assert in 
FTSReplicationStatusUpdateForWalState() making the test more reliable.

### CODE REVIEW

**Comment by RekGRpth** [comment]
Format PR description, please

**Comment by dkovalev1** [comment]
> Format PR description, please

Done

**Comment by RekGRpth** [comment]
Update PR title, please. Hm... title is on two lines?

**Comment by dkovalev1** [comment]
> Update PR title, please. Hm... title is on two lines?

updated again

**Comment by RekGRpth** [comment]
> > Update PR title, please. Hm... title is on two lines?
> 
> updated again

title too, please

File: src/backend/replication/gp_replication.c [hunk 1]
```diff
@@ -181,7 +181,7 @@ FTSReplicationStatusDrop(const char* app_name)
  * FTSReplicationStatusLock should be held before call this function.
  */
 FTSReplicationStatus *
-RetrieveFTSReplicationStatus(const char *app_name, bool skip_warn)
+RetrieveFTSReplicationStatus(const char *app_name, bool skip_err)
```
Comment (RekGRpth) [inline src/backend/replication/gp_replication.c]: change header file too

**Comment by dkovalev1** [comment]
> > > Update PR title, please. Hm... title is on two lines?
> > 
> > 
> > updated again
> 
> title too, please

okay

**Comment by RekGRpth** [comment]
I think it's not good to throw an error while handling another error.

**Comment by RekGRpth** [comment]
Actualize PR description

**Comment by dkovalev1** [comment]
> Actualize PR description

okay

**Review by RekGRpth — APPROVED** [review]

**Review by hilltracer — APPROVED** [review]
I also think that assert is unnecessary. I used the following diagram to understand how the test (case 1 only) works. Maybe it will be useful to someone.
max_slot_wal_keep_size drawio

**Pull request merged**